src/plugins/files/server/file_client/file_metadata_client/adapters/es_index.ts

```diff
@@ -43,7 +44,8 @@ export class EsIndexFilesMetadataClient<M = unknown> implements FileMetadataClie
   constructor(
     private readonly index: string,
     private readonly esClient: ElasticsearchClient,
-    private readonly logger: Logger
+    private readonly logger: Logger,
+    private readonly indexIsAlias: boolean = false
```

Comment:
nit: we usually try to follow the pattern of passing logger last. I understand this prevents `indexIsAlias` from using default and would make the refactor a little larger. Best solution may be proper DI 😄